streams/transform-streams/general.any.js

```diff
@@ -388,9 +388,24 @@ promise_test(t => {
   controller.terminate();
   return Promise.all([
     cancelPromise,
-    promise_rejects_exactly(t, cancelReason, ts.writable.getWriter().closed, 'closed should reject with cancelReason')
+    promise_rejects_js(t, TypeError, ts.writable.getWriter().closed, 'closed should reject with TypeError')
```

Comment:
Hmm, I see. Since `TransformStreamDefaultSourceCancelAlgorithm` takes at least one microtask to run `transformer.cancel()` (even if it does not exist), the `TypeError` from `terminate()` now wins the race. 🤔

I suppose that's fine.

Reply:
hello @MattiasBuelens i am trying to implement this for node this test presently fails my knowledge of WPTs is little limited hence asking here so is `cancelPromise` also expected to throw in this test? I am seeing that running this test standalone throws TypeError but unable to understand why the WPT says this uncaught

Reply:
This test will fail if you have not implanted the changes from the streams spec PR (yet).

Reply:
This test passes against the reference implementation, so if it fails it may be a bug in your implementation.

Reply:
Could be a bug no doubt i just wnated to know this test is basically equalt to the following script right:

```mjs
import { TransformStream } from 'stream/web';

let controller;
const ts = new TransformStream({
  start(c) {
    controller = c;
  }
});
const cancelReason = { name: 'cancelReason' };
const cancelPromise = ts.readable.cancel(cancelReason);
controller.terminate();

try {
  await Promise.all([cancelPromise, ts.writable.getWriter().closed]);
} catch (err) {
  console.log(err); // TypeError!
}
```

Reply:
@debadree25 Yes, that looks script right. Ideally, you want to assert that the `TypeError` originates from `.closed` and not from `cancelPromise`, so in Node you could use [assert.rejects()](https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message):
```javascript
import assert from 'node:assert/strict';

await Promise.all([
  cancelPromise,
  assert.rejects(ts.writable.getWriter().closed, TypeError)
]);
```

> so is `cancelPromise` also expected to throw in this test?

No, `cancelPromise` should always resolve, because `controller.terminate()` calls `ReadableStreamDefaultControllerClose()` which *immediately* closes the readable end.

The only change is that `writer.closed` now becomes errored by `controller.terminate()`, whereas previously it would be errored by `ts.readable.cancel(cancelReason)`. This is because `readable.cancel()` now needs to go through the (possibly asynchronous) `transformer.cancel()` method first before it can error the writable, whereas `controller.terminate()` always errors the writable synchronously (without going through a transformer method).

Reply:
Thank you so much @MattiasBuelens for the detail response i guess i have found the issue with my implementation!